### PR TITLE
Better SymbolValue Helpers

### PR DESCRIPTION
**Pull request by MikeStall**
Add helper for creating SymbolValues from CheckResults. 
This is tricky since it needs to include whatever symbolVAlues the checkresult already has. So Check.Symbols.CreateValues() may miss those and end up creating duplicate symbol values. 

Also add more tests

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/ParsedExpression.cs [hunk 1]
```diff
@@ -91,6 +91,29 @@ internal static IExpressionEvaluator GetEvaluator(this CheckResult result, Stack
 
             return expr;
         }
+
+        /// <summary>
+        /// Create SymbolValues that match against <see cref="CheckResult.Symbols"/>.
+        /// </summary>
+        /// <param name="check"></param>
+        /// <returns></returns>
+        public static ReadOnlySymbolValues CreateValues(this CheckResult check)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/ParsedExpression.cs:100]: This is useful method to add

File: src/libraries/Microsoft.PowerFx.Interpreter/Environment/ComposedReadOnlySymbolValues.cs [hunk 2]
```diff
@@ -91,18 +104,24 @@ internal class ComposedReadOnlySymbolValues : ReadOnlySymbolValues
...
-            catch
+
+            if (map.TryGetValue(symTable, out var existing))
             {
-                // Give better error.
-                if (map.TryGetValue(symTable, out var existingSymValues))
+                if (object.ReferenceEquals(existing, symValues))
                 {
-                    throw new InvalidOperationException($"SymbolTable {symTable.DebugName()} already has SymbolValues '{existingSymValues.DebugName}' associated with it. Can't add '{symValues.DebugName}'");
+                    // Ok, same instance already added
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Environment/ComposedReadOnlySymbolValues.cs:112]: This is fix. 
If it's literally same symbolvalue instance, it's ok

File: src/libraries/Microsoft.PowerFx.Interpreter/ParsedExpression.cs (hunk 1 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Interpreter/ParsedExpression.cs:100]: Love it!

**Review by jas-valgotar — APPROVED** [review]
:shipit:

**Pull request merged**